GAE/src/test/java/org/akvo/flow/xml/FlowXmlObjectReaderTests.java

```diff
@@ -0,0 +1,212 @@
+            + "<question order=\"9\" type=\"option\" mandatory=\"true\" localeNameFlag=\"false\" id=\"43002176\"><options allowOther=\"false\" allowMultiple=\"false\" renderType=\"radio\"><option value=\"Baik\"><text>Baik</text></option><option value=\"Sedang\"><text>Sedang</text></option><option value=\"Kurang\"><text>Kurang</text></option></options><text>Bagaimana kuantitas, kualitas dan tekanan air?</text></question>"
+            + "<question order=\"10\" type=\"option\" mandatory=\"true\" localeNameFlag=\"false\" id=\"43062166\"><options allowOther=\"false\" allowMultiple=\"false\" renderType=\"radio\"><option value=\"Ya \"><text>Ya </text></option><option value=\"Tidak\"><text>Tidak</text></option></options><text>Apakah anda puas dengan pemasangan SR</text></question>"
+            + "</questionGroup>"
+            + "<questionGroup><heading>C. Foto</heading>"
+            + "<question order=\"1\" type=\"photo\" mandatory=\"true\" localeNameFlag=\"false\" id=\"43092142\"><text>Foto Rumah</text></question>"
+            + "<question order=\"2\" type=\"photo\" mandatory=\"true\" localeNameFlag=\"false\" id=\"43032199\"><text>Foto Meter dan air mengalir dari kran  </text></question>"
+            + "<question order=\"3\" type=\"photo\" mandatory=\"true\" localeNameFlag=\"false\" id=\"43062167\"><text>Rekening air 2 bulan</text></question></questionGroup>"
+            + "<questionGroup><heading>D. Hasil Verifikasi</heading>"
+            + "<question order=\"1\" type=\"option\" mandatory=\"false\" localeNameFlag=\"false\" id=\"43042143\"><options allowOther=\"false\" allowMultiple=\"false\" renderType=\"radio\"><option value=\"Memenuhi syarat\"><text>Memenuhi syarat</text></option><option value=\"Belum\"><text>Belum</text></option></options><text>Hasil Verifikasi</text></question>"
+            + "<question order=\"2\" type=\"option\" mandatory=\"true\" localeNameFlag=\"false\" id=\"42842457\"><dependency answer-value=\"Belum\" question=\"43042143\"/><options allowOther=\"false\" allowMultiple=\"false\" renderType=\"radio\"><option value=\"Pemasangan meter air tidak sesuai spesifikasi\"><text>Pemasangan meter air tidak sesuai spesifikasi</text></option><option value=\"Meter air tidak menggunakan SNI\"><text>Meter air tidak menggunakan SNI</text></option><option value=\"Air tidak mengalir\"><text>Air tidak mengalir</text></option><option value=\"Tidak memiliki bukti rekening 2 bulan terakhir \"><text>Tidak memiliki bukti rekening 2 bulan terakhir </text></option></options><text>Alasan belum</text></question>"
+            + "</questionGroup>"
+            + "</survey>";
+
+    @Test
+    void testParseValidForm() throws IOException {
+
+        SurveyDto testFormDto = PublishedForm.parse(MINIMAL_XML_FORM, true).toDto(); //be strict
```

Comment:
I would suggest to test the same way we use it in production, i.e. for now I didn't see any use cases with strict parsing.

Reply:
I thought using the most restrictive case would stress both the code and the test itself better.

Reply:
The strict case if mostly a forward-looking thing. Once we *generate* XML we want to parse it back again in a test to see that is meets the strict definition.

Reply:
I think even in future cases, unless we change how we want things to operate, we want the code to continue working without breaking for example if a new element or attribute is introduced.  And the only consequence is that it is ignored.  Its the same way we do in the app right now. If an attribute is introduced that is not known its ignored but the known ones are parsed correctly.  This keeps it forwards compatible.